miden-tx/src/host/mod.rs

```diff
@@ -51,27 +54,40 @@ pub struct TransactionHost<A> {
     /// The list of notes created while executing a transaction.
     output_notes: Vec<OutputNote>,
 
+    /// Provides a way to get a signature for a message into a transaction
+    tx_authenticator: &'a Option<T>,
```

Comment:
nit: I would probably name this just `authenticator`.

Also, I'm wondering whether instead of using lifetimes we could use smart pointers. Specifically, set the type of `authenticator` to something like `Option<Arc<T>>`. This would also imply that we'd change the `TransactionExecutor` to look something like:
```Rust
pub struct TransactionExecutor<D, A> {
    data_store: D,
    authenticator: Option<Arc<A>>,
    compiler: TransactionCompiler,
    exec_options: ExecutionOptions,
}
```
As far as I can tell, this does not need to change the external interface `TransactionExecutor` - so, may not be a complicated change.